src/Security/MemberAuthenticator/LoginHandler.php

```diff
@@ -263,7 +264,13 @@ protected function redirectToChangePassword()
             'good'
         );
         $changedPasswordLink = Security::singleton()->Link('changepassword');
+        $changePasswordUrl = $this->addBackURLParam($changedPasswordLink);
+        $whitelistedUrl = Controller::join_links($changedPasswordLink, 'ChangePasswordForm');
 
-        return $this->redirect($this->addBackURLParam($changedPasswordLink));
+        $session = $this->getRequest()->getSession();
+        $passwordExpirationMiddleware = Injector::inst()->get(PasswordExpirationMiddleware::class);
+        $passwordExpirationMiddleware->setRedirect($session, $changePasswordUrl);
```

Comment:
This block of code is exactly the same as in the subclass, only without the `allowCurrentRequest`. Do you think `LoginHandler` should have a `getPasswordExpirationMiddleware` function to consolidate this?

Reply:
I still think you can get rid of this code, and let it rely on config

Reply:
> I still think you can get rid of this code, and let it rely on config

Yeah, unfortunately we cannot completely get rid of it and keep CMSLoginHandler working. However, we might get rid of it in LoginHandler though. I'll have a look if we can redo this keeping in mind that we want to reduce coupling between login handlers and the new middleware...

Reply:
Done. We still have to keep it in CMSLoginHandler though.

Reply:
Ok, it broke something. Looking into it

Reply:
Fixed, added a new test